tests/mock.rs

```diff
@@ -0,0 +1,98 @@
+use ic_cdk::api::management_canister::http_request::{
+    CanisterHttpRequestArgument, HttpHeader, HttpMethod, HttpResponse,
+};
+
+pub struct MockOutcallBody(pub Vec<u8>);
+
+impl From<String> for MockOutcallBody {
+    fn from(string: String) -> Self {
+        string.as_bytes().to_vec().into()
+    }
+}
+impl<'a> From<&'a str> for MockOutcallBody {
+    fn from(string: &'a str) -> Self {
+        string.to_string().into()
+    }
+}
+impl From<Vec<u8>> for MockOutcallBody {
+    fn from(bytes: Vec<u8>) -> Self {
+        MockOutcallBody(bytes)
+    }
+}
+
+pub struct MockOutcallBuilder(MockOutcall);
+
+impl MockOutcallBuilder {
+    pub fn new(status: u16, body: impl Into<MockOutcallBody>) -> Self {
+        Self(MockOutcall {
+            method: None,
+            url: None,
+            request_headers: None,
+            request_body: None,
+            response: HttpResponse {
+                status: status.into(),
+                headers: vec![],
+                body: body.into().0,
+            },
+        })
+    }
+
+    pub fn expect_method(mut self, method: HttpMethod) -> Self {
```

Comment:
What's the reason for the `expect_` prefix?

Reply:
Having read the rest of the PR, it looks like the reason is that you're building up the expected struct in tests, correct?
Since it's simply an application of the builder pattern, I would use a more generic prefix like `set_`, `with_`, or something like that.

Reply:
I've started to come down with (probably) COVID and inexplicably changed a lot of the names away from these more reasonable suggestions. Was originally `with_url`, `with_request_body`, etc. Totally agreed that it's clearer that way.